.github/workflows/testing.yml

```
@@ -39,25 +39,27 @@ jobs:
       fail-fast: false
       matrix:
         python-version: [ "3.7", "3.8" , "3.9", "3.10" ]
-        django-version: [ "3.0", "3.1", "3.2", "4.0" ]
+        django-version: [ "3.0", "3.1", "3.2", "4.0", "4.1" ]
         exclude:
           # Django v4 drops Python 3.6, and 3.7 support
           - django-version: 4.0
             python-version: 3.7
+          - django-version: 4.1
+            python-version: 3.7
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-python@v2
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
         with:
           python-version: "${{ matrix.python-version }}"
-      - uses: actions/cache@v2
+      - uses: actions/cache@v3
         id: poetry-cache
         with:
           path: ~/.local
           key: key-3
```

When touching this file, let's update the `key` name to something bringing more context and is dependent on the `python-version` matrix variable (this will speed up also CI pipeline):

```suggestion
          key: "poetry-cache-python${{ matrix.python-version }}"
```

---

Is the Poetry installation really Python version specific though? I'm not so sure.

How will it speed up the pipeline?

---

`poetry` shouldn't be python version specific, but dependencies could be, that's the reason cache key should be parametrized with python version

---

This cache only caches the poetry installation 🙂

---

Ahh, ok my bad, sorry!
Because `virtualenvs` location-related option (`virtualenvs.in-project
` or `virtualenvs.path`) wasn't set, I thought it will create `.venv` in the default location, but according to docs about [`virtualenvs.create`](https://python-poetry.org/docs/configuration/#virtualenvscreate) poetry will install dependencies in currently active env